### PR TITLE
Recipe refactor tj

### LINKED ISSUE #2: Check current Recipe functionality
Users should be able to view a list of recipes. Specifically:

- [x] As a user, I should be able to filter recipes by type / tag
- [x] As a user, I should be able to search recipes by ingredients

A recipe should hold on to all its information (provided in the data file). It should be able to:

- [x] Get the cost of its ingredients
- [x] Get its directions / instructions

- [x] All required methods are present
- [x] All required properties are present

### LINKED ISSUE #19: Recipes - Filter Recipes
As a user, I should be able to filter the list of viewed recipes by type/tag

### LINKED ISSUE #21: Recipes - Recipe information
A recipe should hold on to all its information (provided in the data file)

### LINKED ISSUE #24: Recipes - Ingredient cost
The recipes class should have a method to get the cost of ingredients

### LINKED ISSUE #26: Recipes - Get instructions
The Recipes class should be able to get the instructions/ recipe directions

### PR DESCRIPTION
**Pull request by taylorjohnson141**
## Description

Clearly and concisely describe the feature.
This feature gives the Recipe.js Class the ability to
Search by a given ingredient
filter tags by given letters
Return current instructions 

## Affected areas of application

This PR will add the following methods to the Recipe.js

returnInstructions
filterByTag
containsIngredient
searchByIngredient

This PR also tests these methods in test/recipe-test.js

List all parts of the website that will be altered from this PR.
No parts of the website have been directly effected by the PR
## How Has This Been Tested?

The Pr was tested using Chai and Mocha as well as Webpack

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

First run npm install to install the required mocha and chai dependencies as well as webpack
Then Run np

## Relevant Tickets

What tickets/issues were resolved by this feature

closes #2 
closes #19 
closes #21 
closes #24 
closes #26 

## Screenshots

Add screenshots here if applicable

## Future Considerations
The way I have my current search methods set up return the values they are looking for. It might be better to return a boolean but I am not sure.

### CODE REVIEW

File: src/recipe.js [hunk 1]
```diff
@@ -20,7 +20,34 @@ class Recipe {
...
   }
-
+  returnInstructions() {
+    return this.instructions
+  }
+  filterByTag(letter) {
+    var tags =  this.tags.filter(tag =>{
+      return tag.split('').includes(letter.toLowerCase())
+    })
+    console.log(tags)
```
Comment (nicolegooden) [inline src/recipe.js:30]: Make sure you remove this `console.log()` at some point :)

File: test/recipe-test.js [hunk 2]
```diff
@@ -1,32 +1,148 @@
 import { expect } from 'chai';
-
 import Recipe from '../src/recipe.js';
-import recipeData from '../src/data/recipes.js';
-import ingredientsData from '../src/data/ingredients.js';
-
+let recipeData =
```
Comment (CalebCyphers) [inline test/recipe-test.js:3]: Thanks for adding in the self-contained data, this works great!

**Review by CalebCyphers — APPROVED** [review]

File: src/recipe.js [hunk 3]
```diff
@@ -20,7 +20,34 @@ class Recipe {
...
+    return tags
+  }
+  containsIngredient(inputIngredient) {
```
Comment (nicolegooden) [inline src/recipe.js:33]: I like how you added a method here that isn't pointed out in the spec.  Good thinking ahead!

File: src/recipe.js [hunk 4]
```diff
@@ -20,7 +20,34 @@ class Recipe {
     });
     return costCounter;
   }
-
+  returnInstructions() {
+    return this.instructions
+  }
+  filterByTag(letter) {
+    var tags =  this.tags.filter(tag =>{
```
Comment (nicolegooden) [inline src/recipe.js:27]: this method is really clean, nice job!

File: test/recipe-test.js [hunk 5]
```diff
@@ -1,32 +1,148 @@
...
+      "estimatedCostInCents": 742
+    }
+];
 let recipe;
-
 describe('Recipe', () => {
   beforeEach(() => {
 
-    recipe = new Recipe(recipeData[47], ingredientsData);
+    recipe = new Recipe(recipeData, ingredientsData);
```
Comment (nicolegooden) [inline test/recipe-test.js:90]: this is interesting, I didn't realize the datasets could be defined outside of the describe block. Good to know.

**Review by nicolegooden — APPROVED** [review]

**Pull request merged**